src/Microsoft.DotNet.Cli.Utils/CommandResolution/CommandResolverCommon.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.ProjectModel.Graph;
+using Microsoft.Extensions.PlatformAbstractions;
+using NuGet.Frameworks;
+using NuGet.Packaging;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    internal class CommandResolverCommon
+    {
+        public static CommandSpec CreateCommandSpecPreferringExe(
+           string commandName,
+           IEnumerable<string> args,
+           string commandPath,
+           CommandResolutionStrategy resolutionStrategy,
+           IEnvironmentProvider environment)
+        {
+            var useComSpec = false;
+
+            if (PlatformServices.Default.Runtime.OperatingSystemPlatform == Platform.Windows &&
```

Comment:
This if here is a indicative that you should have different implementations for this. Then you won't need this. You will have a WindowsCommandResolverCommon that would have this in there.